backend/app/analysis/proximity.py

```diff
@@ -0,0 +1,88 @@
+import nltk
+from collections import Counter
+from more_itertools import windowed
+
+from ..models import (
+    Document,
+    Gender,
+    Corpus,
+)
+
+
+def run_analysis(corpus_id, gender_ids, word_window):
```

Comment:
I think we can simply pass the `Corpus` and `Gender` instances here. I can imagine that there'll be an API layer that performs some basic validation to ensure that the `Gender` and `Corpus` instances specified by the API input (say, a user passes down the appropriate `id` for each from the front end) actually do exist. That validation will necessarily require the look up for each of these instances, so passing them down here will save us using `Corpus.objects.filter` below (i.e., it can be come `corpus.filter(etc...)`. We can also pass in the `Gender` instances instead of their `id`s, turning `Gender.objects.get` into simply `gender` (vel sim.).

Reply:
@MBJean Our team was actually wondering about this. An the `transfer_proximity` branch, we created a new `ProximityAnalyses` model to store the results of these analyses in the database and used it as a sort of caching mechanism in `proximity.py` (we're about to make a PR). In that branch, we modified the three functions this PR presents, and one of the modifications was scrapping the `gender_ids` paramater. We figured it might be better to run the analysis for all `Gender` objects in the database and leave it to frontend to filter the display as per the user's preferences. I've changed the code so that we now pass in a set of all `Gender` objects instead of the list of `gender_ids`.

Regarding the `Corpus` object, we were unsure if loading the `Corpus` object would also load all associated `Document` objects from the database. Without the `Document`s, each `Corpus` object is pretty slim, so we could probably also implement that validation (possibly using the [`.exists()`](https://docs.djangoproject.com/en/3.1/ref/models/querysets/#exists) query) in a view function.

Reply:
Update about that new PR I mentioned: with these new changes, we'll likely have to go back and rethink our implementation for the other functions in `proximity.py` as well as how we will connect this to an API endpoint.